composeApp/src/commonMain/kotlin/presentation/screens/feed/FeedScreen.kt

```diff
@@ -0,0 +1,270 @@
+                    openPost(it.id)
+                },
+                onShare = {
+                    onSharePost(it)
+                }
+            )
+        }
+    }
+}
+
+@Composable
+fun ItemPost(
+    post: Post,
+    onClick: () -> Unit = {},
+    onShare: () -> Unit = {}
+) {
+    PostHeader(post = post)
+    Spacer(modifier = Modifier.height(8.dp))
+    if (post.images.size > 1) {
+        PostMedia(images = post.images) {
+            onClick()
+        }
+    }
+    PostActions(
+        post = post,
+        onShare = onShare
+    )
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun PostHeader(post: Post) {
+    ListItem(
+        text = {
+            Text(
+                text = post.user.name,
+                fontWeight = FontWeight.Medium,
+            )
+        },
+        secondaryText = {
+            Text(
+                text = post.user.username,
+                style = MaterialTheme.typography.body2,
+            )
+        },
+        trailing = {
+            Icon(
+                imageVector = Icons.Outlined.MoreVert,
+                contentDescription = "More options",
+            )
+        },
+        icon = {
+            Box(
+                modifier = Modifier
+                    .size(40.dp)
+                    .clip(CircleShape)
+                    .background(Color.Gray)
+            ) {
+                AsyncImage(
+                    model = post.user.avatar,
+                    contentDescription = "Avatar",
+                    modifier = Modifier.fillMaxSize(),
+                    contentScale = ContentScale.Crop
+                )
+            }
+        }
+    )
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun PostMedia(
+    images: List<String>,
+    onClick: () -> Unit
+) {
+
+    val pagerState = rememberPagerState(pageCount = {
+        images.size
+    })
+
+    HorizontalPager(state = pagerState) { page ->
+        Box(
+            modifier = Modifier
+                .background(Color.Gray)
+                .fillMaxWidth()
+                .height(200.dp)
+                .clickable { onClick() },
+            contentAlignment = Alignment.TopEnd
+        ) {
+            AsyncImage(
+                model = images[page],
+                contentDescription = "Post image",
+                modifier = Modifier.fillMaxSize(),
+                contentScale = ContentScale.Crop
+            )
+
+            Box(
+                modifier = Modifier
+                    .padding(8.dp)
+                    .clip(CircleShape)
+                    .background(Color.Black.copy(alpha = 0.5f))
+            ) {
+                Text(
+                    text = "${page + 1} / ${images.size}",
+                    color = Color.White,
+                    fontSize = 12.sp,
+                    modifier = Modifier.padding(12.dp, 8.dp),
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun PostActions(
```

Comment:
**#suggestion**

We can move this component to be a new file inside of a components package, just to keep the Screen file smaller.